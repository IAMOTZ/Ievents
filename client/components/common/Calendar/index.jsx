import React from 'react';
import PropTypes from 'prop-types';
import InfiniteCalendar from 'react-infinite-calendar';
import 'react-infinite-calendar/styles.css';

const Calendar = (props) => {
  const today = new Date();
  const tommorrow = new Date(today.getFullYear(), today.getMonth(), today.getDate() + 1);
  let minDate;
  // There must be 24hours between today and the minimum date that is selectable.
  if (tommorrow.getTime() - today.getTime() >= 24 * 60 * 60 * 1000) {
    minDate = tommorrow;
  } else {
    minDate = new Date(today.getFullYear(), today.getMonth(), today.getDate() + 2);
  }
  const maxMonth = new Date(today.getFullYear(), today.getMonth() + 12, today.getDate());
  const minMonth = new Date(today.getFullYear(), today.getMonth(), today.getDate());
  const selectedDate = props.selectedDate ? new Date(props.selectedDate) : false;

  return (
    <InfiniteCalendar
      width={400}
      height={300}
      selected={selectedDate}
      minDate={minDate}
      max={maxMonth}
      min={minMonth}
      onSelect={props.handleDateSelection}
      displayOptions={{
        showTodayHelper: false
      }}
      theme={{
        selectionColor: 'rgb(0, 142, 214)',
      }}
    />
  );
};

Calendar.defaultProps = {
  selectedDate: null,
};

Calendar.propTypes = {
  handleDateSelection: PropTypes.func.isRequired,
  selectedDate: PropTypes.string,
};

export default Calendar;
