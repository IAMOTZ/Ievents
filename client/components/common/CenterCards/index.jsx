import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './styles.scss';

const defaultImage = '/images/defaultImgx4.jpeg';

const CenterCards = props => props.centers.map(center => (
  <div className="col-lg-4 col-md-6 col-sm-1" key={center.id}>
    <div className="card centerCard">
      <div>
        <img className="card-img-top" src={center.images ? center.images[0] : defaultImage} alt="The Center" />
        <div className="badge-container">
          <span className="badge text-white p-2 seat-badge ">{center.capacity} seats</span>
        </div>
      </div>
      <div className="card-body d-flex flex-column">
        <div>
          <h4 className="card-title text-capitalize">{center.name}</h4>
          <div className="icons">
            <div>
              <i className="fa fa-map-marker fa-fw" aria-hidden="true" />&nbsp;
              <span className="text-capitalize">{center.location}</span>
            </div>
            <div>
              <i className="fa fa-info-circle fa-fw" aria-hidden="true" />&nbsp;
              <a data-toggle="collapse" href={`#center-details-collapse-${center.id}`} role="button">About center</a>
            </div>
          </div>
          <div className="collapse" id={`center-details-collapse-${center.id}`}>
            <br />
            <span className="center-details">{center.details}</span>
          </div>
        </div>
        {
          props.isAdmin ?
            <div className="admin-only-icons">
              <i className="fa fa-pencil fa-fw" aria-hidden="true" />&nbsp;
              <Link
                to="/editCenter"
                id={center.id}
                onClick={props.onEdit}
              >Edit Center
              </Link>
            </div> : null
        }
        <Link
          to="/addEvent"
          onClick={props.onBook}
          id={center.id}
          className="btn ie-blue-button mt-2 btn-block"
        > Book now for â‚¦{center.price}/Day
        </Link>
      </div>
    </div>
  </div>
));

/* eslint-disable react/forbid-prop-types */
CenterCards.propTypes = {
  centers: PropTypes.array.isRequired,
  onBook: PropTypes.func.isRequired,
  onEdit: PropTypes.func,
};

export default CenterCards;
