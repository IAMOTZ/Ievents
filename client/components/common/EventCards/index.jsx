import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';
import {
  RenderEventBody,
  RenderEventBodyInTransactions,
  RenderEventFooter
} from './subComponents';

/* eslint-disable react/forbid-prop-types */
const EventCards = props => props.events.map(event => (
  <div className="col-lg-4 col-md-6 col-sm-1" key={event.id}>
    <div
      style={{ maxWidth: `${20}rem` }}
      className="card text-dark bg-white mb-3 mx-auto event-card"
    >
      <div className="card-header">
        <span className="h4 mb-0">{event.title}</span>
      </div>
      {
        props.isTransactionsPage ?
          <RenderEventBodyInTransactions
            event={event}
            createModalContent={props.createModalContent}
            startEventCancel={props.startEventCancel}
            cancelingTransactionStarted={props.cancelingTransactionStarted}
          /> :
          <RenderEventBody
            event={event}
            createModalContent={props.createModalContent}
          />
      }
      {
        props.isTransactionsPage ?
          null :
          <RenderEventFooter
            event={event}
            startDelete={props.startDelete}
            edit={props.edit}
          />
      }
    </div>
  </div>
));
EventCards.propTypes = {
  events: PropTypes.array,
};
EventCards.defaultProps = {
  events: []
};


export default EventCards;
