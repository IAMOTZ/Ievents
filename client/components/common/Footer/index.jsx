import React from 'react';

const footerStyle = {
  backgroundColor: 'rgba(0,0,0, 0.9)',
  borderTop: '2px solid rgb(0, 142, 214)'
};

const Footer = () => (
  <footer className="w-100 mt-auto" style={footerStyle}>
    <div className="container text-white text-center py-5">
      <h1>iEvents</h1>
      <p>Copyright &copy; 2017</p>
    </div>
  </footer>
);

export default Footer;
