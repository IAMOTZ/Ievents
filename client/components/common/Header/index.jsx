import React from 'react';
import PropTypes from 'prop-types';

const headerStyle = {
  backgroundColor: 'rgba(0,0,0, 0.9)'
};

const Header = props => (
  <nav className="navbar w-100 mt-3 d-none d-lg-block" style={headerStyle} id="header">
    <strong className="navbar-brand text-white">{props.text}</strong>
  </nav>
);

Header.defaultProps = {
  text: null,
};
Header.propTypes = {
  text: PropTypes.string,
};

export default Header;
