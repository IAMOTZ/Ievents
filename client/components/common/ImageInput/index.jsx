import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';
import './styles.scss';

const defaultImage = '/images/defaultImgx4.jpeg';

const ImageInput = (props) => {
  const { newImage, previousImage } = props;
  let captionText;
  let imageSource;
  if (newImage) {
    imageSource = newImage;
    captionText = 'Click to select another image.';
  } else if (previousImage) {
    imageSource = previousImage;
    captionText = 'Click to select another image.';
  } else {
    imageSource = defaultImage;
    captionText = 'This is the default center image. Click to select another image.';
  }
  return (
    <Dropzone
      className="react-dropzone"
      multiple={false}
      accept=".jpg, .jpeg, .png"
      onDropAccepted={props.onDrop}
    >
      <div className="image-preview">
        <img src={imageSource} alt="" />
        <div className="image-preview-overlay">
          <span>{captionText}</span>
        </div>
      </div>
    </Dropzone>
  );
};

ImageInput.defaultProps = {
  newImage: null,
  previousImage: null,
};

ImageInput.propTypes = {
  onDrop: PropTypes.func.isRequired,
  newImage: PropTypes.string,
  previousImage: PropTypes.string,
};

export default ImageInput;
