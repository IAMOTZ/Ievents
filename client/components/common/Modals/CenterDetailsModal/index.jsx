import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

const defaultImage = '/images/defaultImgx4.jpeg';

const CenterDetailsModal = props => (
  <div className="modal fade" tabIndex="-1" role="dialog" id="center-details-modal">
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title text-capitalize">{props.centerName}</h5>
          <button
            type="button"
            className="close no-outline-btn pointer-button"
            data-dismiss="modal"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          <div className="center-image">
            <img src={props.centerImages ? props.centerImages[0] : defaultImage} alt="The Center" />
          </div>
          <div className="center-details">
            <p>{props.centerDetails}</p>
          </div>
        </div>
      </div>
    </div>
  </div>
);

CenterDetailsModal.defaultProps = {
  centerName: '',
  centerDetails: '',
  centerImages: [defaultImage],
};

/* eslint-disable react/forbid-prop-types */
CenterDetailsModal.propTypes = {
  centerName: PropTypes.string,
  centerDetails: PropTypes.string,
  centerImages: PropTypes.array,
};

export default CenterDetailsModal;
