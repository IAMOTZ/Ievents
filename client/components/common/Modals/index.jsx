import CenterModal from './CenterModal';
import ConfirmationModal from './ConfirmationModal';

export { CenterModal, ConfirmationModal };
