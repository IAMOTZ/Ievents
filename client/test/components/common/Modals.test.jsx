import React from 'react';
import { ConfirmationModal, CenterDetailsModal, EventDetailsModal } from '../../../components/common/Modals';

describe('<ConfirmationModal />', () => {
  const props = {
    onCancel: jest.fn(),
    onOK: jest.fn(),
  };
  let wrapper;
  beforeEach(() => {
    const childComponent = <p>I am the confirmation modal</p>;
    wrapper = shallow(<ConfirmationModal {...props}>{childComponent}</ConfirmationModal>);
  });
  describe('Rendering:', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Interactions:', () => {
    it('should call the function attached to the `No` button', () => {
      wrapper.find('button.btn.ie-dark-button').simulate('click');
      expect(props.onCancel.mock.calls.length).toEqual(1);
    });
    it('should call the function attached to the `Yes` button', () => {
      wrapper.find('button.btn.ie-blue-button').simulate('click');
      expect(props.onOK.mock.calls.length).toEqual(1);
    });
  });
});

describe('<CenterDetailsModal />', () => {
  const props = {
    centerName: 'Test Center',
    centerDetails: 'I am the test center',
    centerImages: null,
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CenterDetailsModal {...props} />);
  });
  describe('Rendering:', () => {
    it('should use default image when centerImages is not passed in props', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should use the image passed as centerImages in props', () => {
      wrapper.setProps(alterProps({ centerImages: ['https://linkToCenterImage'] }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('<EventDetailsModal />', () => {
  const props = {
    eventTitle: 'Test event',
    eventDescription: 'I am the test event',
  };
  describe('Rendering:', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<EventDetailsModal {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
