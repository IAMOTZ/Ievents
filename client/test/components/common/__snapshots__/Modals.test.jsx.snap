// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterDetailsModal /> Rendering: should use default image when centerImages is not passed in props 1`] = `
<div
  className="modal fade"
  id="center-details-modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title text-capitalize"
        >
          Test Center
        </h5>
        <button
          className="close no-outline-btn pointer-button"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="center-image"
        >
          <img
            alt="The Center"
            src="/images/defaultImgx4.jpeg"
          />
        </div>
        <div
          className="center-details"
        >
          <p>
            I am the test center
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CenterDetailsModal /> Rendering: should use the image passed as centerImages in props 1`] = `
<div
  className="modal fade"
  id="center-details-modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title text-capitalize"
        >
          Test Center
        </h5>
        <button
          className="close no-outline-btn pointer-button"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="center-image"
        >
          <img
            alt="The Center"
            src="https://linkToCenterImage"
          />
        </div>
        <div
          className="center-details"
        >
          <p>
            I am the test center
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ConfirmationModal /> Rendering: should render correctly 1`] = `
<div
  className="modal fade"
  id="confirmation-modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Confirmation
        </h5>
        <button
          className="close pointer-button"
          data-dismiss="modal"
          onClick={[MockFunction]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
          I am the confirmation modal
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn ie-dark-button"
          data-dismiss="modal"
          onClick={[MockFunction]}
          type="button"
        >
          No
        </button>
        <button
          className="btn ie-blue-button"
          onClick={[MockFunction]}
          type="button"
        >
          Yes
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<EventDetailsModal /> Rendering: should render correctly 1`] = `
<div
  className="modal fade"
  id="event-details-modal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Test event
        </h5>
        <button
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
          I am the test event
        </p>
      </div>
    </div>
  </div>
</div>
`;
